src/PactNet.Native/NativePactBuilder.cs

```diff
@@ -63,10 +62,58 @@ internal NativeRequestBuilder UponReceiving(string description)
         }
 
         /// <summary>
-        /// Finalise the pact
+        /// Verify the configured interactions
+        /// </summary>
+        /// <param name="interact">Action to perform the real interactions against the mock server</param>
+        /// <exception cref="PactFailureException">Failed to verify the interactions</exception>
+        public void Verify(Action<IConsumerContext> interact)
+        {
+            Uri uri = this.StartMockServer();
+
+            try
+            {
+                interact(new NativeConsumerContext
```

Comment:
I would use interact?.Invoke(...)
Or check the interact action is not null before and throw an ArgumentNullException

Reply:
Added a null check